patches/ios-chrome-browser-net-ios_chrome_network_delegate.h.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/ios/chrome/browser/net/ios_chrome_network_delegate.h b/ios/chrome/browser/net/ios_chrome_network_delegate.h
+index 830e12ec3a388..00ed8ecbb6083 100644
+--- a/ios/chrome/browser/net/ios_chrome_network_delegate.h
++++ b/ios/chrome/browser/net/ios_chrome_network_delegate.h
+@@ -48,6 +48,7 @@ class IOSChromeNetworkDelegate : public net::NetworkDelegateImpl {
+                                         PrefService* pref_service);
+ 
+  private:
++  friend IOSChromeNetworkDelegate_BraveImpl;
```

Comment:
you can do this without patching
```
#define OnBeforeURLRequest NotUsed() { return 0; }; friend IOSChromeNetworkDelegate_BraveImpl; OnBeforeURLRequest
```
but if you couldn't the convention would be to add BRAVE_IOS_CHROME_NETWORK_DELEGATE_H so the patch would be extensible

Reply:
@bridiver We actually had it similar to that originally but I noticed many other patches that just add friend classes like above so figured we'd want to do that as well. If you want me to remove the patch and update the chromium_src override to use preprocessor macro overrides instead, let me know and I can commit that :)

Reply:
the patches for `friend class` are old. The preferred method is `#define OnBeforeURLRequest ...` or `BRAVE_IOS_CHROME_NETWORK_DELEGATE_H` if that isn't possible

Reply:
if you saw recent patches with `friend class` please let me know so I can talk to the patch reviewer who approved it